versions/3.0.4.md

```diff
@@ -89,11 +89,11 @@ The OpenAPI Specification is versioned using a `major`.`minor`.`patch` versionin
 
 Occasionally, non-backwards compatible changes may be made in `minor` versions of the OAS where impact is believed to be low relative to the benefit provided.
 
-An OpenAPI description document compatible with OAS 3.\*.\* contains a required [`openapi`](#oas-version) field which designates the version of the OAS that it uses.
+An OpenAPI Description compatible with OAS 3.\*.\* contains a required [`openapi`](#oas-version) field which designates the version of the OAS that it uses.
 
 ### Format
 
-An OpenAPI description document that conforms to the OpenAPI Specification is itself a JSON object, which may be represented either in JSON or YAML format.
+An OpenAPI Description that conforms to the OpenAPI Specification is itself a JSON object, which may be represented either in JSON or YAML format.
```

Comment:
So is this about "An OpenAPI entry document that conforms"? 

Because technically referenced documents could have a top-level array that contains to-be-referenced objects, arrays, or primitive values.

Reply:
The original text "OpenAPI Description" is correct IMO. We already did a lot of work on not using "document" when we mean "description"

Reply:
@lornajane but we _do_ mean "document" here, not "Description."  It is not always possible to represent an entire OAD as a single JSON/YAML object,

@ralfhandl 
> So is this about "An OpenAPI entry document that conforms"?

it's not just about entry documents.  This gets into the distinction I waas trying to make with "syntactically complete" documents.

> Because technically referenced documents could have a top-level array that contains to-be-referenced objects, arrays, or primitive values.

No, such a document does not, _as a document_ conform to the OAS.  Nowhere in this specification are such documents mentioned, much less deemed compliant with the OAS syntax.

You _can_ make a plausible argument that a document with an Object other than an OpenAPI Object at its root is "compliant" with the OAS, although the 3.0 specification never directly says so one way or the other.  I'm not sure 2.0 even says that, although it strongly implies it.

So there are really three cases here:

1. Syntactically complete documents (with an OpenAPI Object at the root) — these are unquestionably in compliance with the OAS, regardless of version
2. Documents with some other Object at the root — these are never explicitly mentioned in 3.x, and the proposed OAS  media types _do not support them_ as currently written, but they definitely work in 3.0.  3.1.1 lists the conditions under which they can work in 3.1.  I think of these as fragments (OAD fragments?  OAS fragments?  compatible with fragments of the OAS?)
3. JSON/YAML documents in arbitrary format where some subset of the document (e.g. each element in a root array) is compliant with some or all of the OAS format.  This is not addressed _anywhere_ in the spec AFAICT.  It can work for 3.0.  But 3.1.1 explicitly states that this sort of behavior is undefined, because you cannot correctly process only part of a 3.1 document (there's a whole section about exactly this in 3.1.1)

Reply:
Personally I have never made a distinction between 2 and 3 before.  I have referred to the chunks of these documents that do conform to some OAS object as OpenAPI fragments.
So, in my head there are documents that are OpenAPI Documents and documents that are JSON/YAML documents that can contain OpenAPI fragments.

Reply:
Changed to "OpenAPI Document", which is now well-defined.